Index: src/test/com/thoughtworks/qdox/AnnotationsTest.java
===================================================================
--- src/test/com/thoughtworks/qdox/AnnotationsTest.java	(revision 515)
+++ src/test/com/thoughtworks/qdox/AnnotationsTest.java	(working copy)
@@ -163,6 +163,14 @@
         builder.addSource(new StringReader(source));
         assertEquals("Foo", builder.getClassByName("Foo").getName());
     }
+    
+    //from QDOX-128
+    public void testQuotedStringAnnotation()  throws Exception {
+    	String source = "@Anno(run = \"1.0\")"+
+    	"public interface Foo {}";
+        builder.addSource(new StringReader(source));
+        assertEquals("\"1.0\"", builder.getClassByName("Foo").getAnnotations()[0].getProperty("run").getParameterValue());
+    }
 
     // from QDOX-135
     public void testAnnotationInMethodParamList() {
